### PR TITLE
Authenticate for both ws and ms

### PR DESCRIPTION
**Pull request by jian-en**

### CODE REVIEW

**Comment by bwan2** [comment]
@jian-en @elibitt there are still .idea files in the commit

**Comment by elibitt** [comment]
We know, Jian is adding a gitignore right now

**Comment by elibitt** [comment]
@bwan2 Should be good to go now!

File: ms/code/AuthenticateServices.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
+/******************************************************************************************************************
+ * File: RetrieveServices.java
```
Comment (innozhang) [inline ms/code/AuthenticateServices.java]: change the java doc comments

File: ms/code/OrdersUI.java [hunk 2]
```diff
@@ -50,9 +52,26 @@ public static void main(String args[])
...
+		{
+			// Pre: authentication process
+			if (!api.authFlag) {
+				System.out.println("Enter your username: \n");
+				username = keyboard.nextLine();
+				System.out.println("Enter your password: \n");
+				password = keyboard.nextLine();
+				try {
+					if (api.authenticateUser(username, password))
+						System.out.println("You are authenticated!! \n");
```
Comment (bwan2) [inline ms/code/OrdersUI.java]: only one '!'; also this string needs to be the same as ws/code/OrdersUi.java 

> 						System.out.println( "Invalid user credentials!\n" );

File: ms/code/OrdersUI.java [hunk 3]
```diff
@@ -50,9 +52,26 @@ public static void main(String args[])
...
+					else {
+						System.out.println("You have input the wrong username/password!! \n");
```
Comment (bwan2) [inline ms/code/OrdersUI.java]: only one '!'; also this string needs to be the same as ws/code/OrdersUi.java

> 					System.out.println("Incorrect User Credentials! Try again.\n");

File: ms/code/AuthenticateServicesAI.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+
```
Comment (bwan2) [inline ms/code/AuthenticateServicesAI.java:24]: add intro comments

File: ms/code/AuthenticateServicesAI.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+import java.rmi.*;
+
+public interface AuthenticateServicesAI extends java.rmi.Remote
+{
+  /*******************************************************
+   * Authenticates user with username and password
```
Comment (bwan2) [inline ms/code/AuthenticateServicesAI.java:30]: this comment doesnt say much, add more details on how its being used

File: ws/server/UserService.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+                 password: 'test'}];
+
+module.exports.authenticate = authenticate;
+
+function authenticate({ username, password }) {
+    const user = users.find(u => u.username === username && u.password === password);
+    console.log(username);
+    console.log(password);
+    if (user) {
+    	console.log("Athenticated good");
```
Comment (bwan2) [inline ws/server/UserService.js]: typo

**Pull request merged**